src/examples/java/software/amazon/encryption/s3/examples/MultipartUploadExample.java

```diff
@@ -0,0 +1,192 @@
+package software.amazon.encryption.s3.examples;
+
+import org.apache.commons.io.IOUtils;
+import software.amazon.awssdk.core.ResponseBytes;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.*;
+import software.amazon.awssdk.utils.IoUtils;
+import software.amazon.encryption.s3.S3EncryptionClient;
+import software.amazon.encryption.s3.utils.BoundedInputStream;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static software.amazon.encryption.s3.S3EncryptionClient.withAdditionalConfiguration;
+import static software.amazon.encryption.s3.utils.S3EncryptionClientTestResources.KMS_KEY_ID;
+import static software.amazon.encryption.s3.utils.S3EncryptionClientTestResources.appendTestSuffix;
+
+public class MultipartUploadExample {
+    public static String BUCKET;
+    public static void main(final String[] args) throws IOException {
+        BUCKET = args[0];
+        LowLevelMultipartUpload();
+        HighLevelMultipartPutObject();
+    }
+
+    public static void LowLevelMultipartUpload() throws IOException {
+        final String objectKey = appendTestSuffix("low-level-multipart-upload-example");
+
+        // Overall "file" is 100MB, split into 10MB parts
+        final long fileSizeLimit = 1024 * 1024 * 100;
+        final int PART_SIZE = 10 * 1024 * 1024;
+        final InputStream inputStream = new BoundedInputStream(fileSizeLimit);
+
+        // Instantiate the S3 Encryption Client to encrypt and decrypt
+        // by specifying a KMS Key with the kmsKeyId builder parameter.
+        // You must also specify the enableDelayedAuthenticationMode parameter to upload more than 64MB object or increase your default buffer size.
```

Comment:
this isn't quite accurate - delayed auth is not required to upload objects larger than 64Mib/configured buffer size, only for download